wordpress-mobile/WordPress-Android
# Use live data in stats use cases

In this PR I'm moving the LiveData sources down to the UseCases. I've introduced a shared interface for all the UseCases which also allows us to clear the value when the ViewModel is torn down.

I've moved all the use cases into a package to increase the readibility.

I've also fixed all the tests. The only thing missing now are the tests for the DB loading of the data. 

To test:
- Open stats
- All the stats block are loaded
- Refreshing the list works
- Scrolling around works
- Change the current site
- Go back to Stats
- You don't see the old stats but only an empty list that gets filled once the data is loaded
